api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -64,6 +64,18 @@ message Filter {
 // ``www.example.com``, then ``*.example.com``, then ``*.com``, then any filter
 // chain without ``server_names`` requirements).
 //
+// Yet another approach to reason about the filter chain matches.
+// Suppose you have N filter chains in hand. Sieve the filter chain set in the above 8 rounds.
+// In each round, filter chains matching the attributes best survives to the next round.
+// The listener guarantees at most 1 filter chain left after all the rounds.
+//
+// Example:
+//
+// For destination port, filter chains specifying the destination port of incoming traffic are the
```

Comment:
So if I have 1 FC with a destination port, and another without destination port set, the latter will never match? And if I do want "everything but destination port from FC1" I need to make 65535 FCs? Or am I reading wrong

Reply:
More concrete example, suppose listener has 4 filter chains
FC1 {dest port 80}
FC 2{dest port 90}
FC 3{dest port: null}
FC 4{dest port: null, ip = 10.0.0.1}

For incoming traffic which has  
a) dest port = 80
    {FC1} is the best match
b) dest port = 90
   {FC2} is the best match
c) dst port = 100
   {FC3, FC4} is the best match